src/apps/Settings/HomeSettings.js

```diff
@@ -0,0 +1,169 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { Button, DropDown, Field, TextInput } from '@aragon/ui'
+
+import { AppType, EthereumAddressType } from '../../prop-types'
+import AragonStorage from '../../storage/storage-wrapper'
+
+import Option from './Option'
+
+const defaultName = 'Use the default'
+
+class HomeSettings extends React.Component {
+  static propTypes = {
+    account: EthereumAddressType,
+    apps: PropTypes.arrayOf(AppType).isRequired,
+    walletWeb3: PropTypes.object.isRequired,
+  }
+  state = {
+    homeAppName: 'Home',
+    selectedHomeAppName: defaultName,
+    storageApp: null,
+  }
+
+  async getHomeSettings(props) {
+    const { apps, walletWeb3, account } = props
+    const storageApp = apps.find(({ name }) => name === 'Storage')
+    let homeAppName, homeAppAddr
+
+    // TODO: Move this code to a subscriber
+    if (storageApp && storageApp.proxyAddress) {
+      homeAppAddr = await AragonStorage.get(
+        walletWeb3,
+        storageApp.proxyAddress,
+        account,
+        'HOME_APP'
+      )
+
+      homeAppName = await AragonStorage.get(
+        walletWeb3,
+        storageApp.proxyAddress,
+        account,
+        'HOME_APP_NAME'
+      )
+      if (!homeAppName) {
+        homeAppName = 'Home'
+      }
+    }
+    const selectedHomeApp = apps.find(
+      ({ proxyAddress }) => proxyAddress === homeAppAddr
+    )
+
+    this.setState({
+      homeAppName: homeAppName,
+      selectedHomeAppName:
+        (selectedHomeApp && selectedHomeApp.name) || defaultName,
+      storageApp: apps.find(({ name }) => name === 'Storage'),
+    })
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.getHomeSettings(nextProps)
+  }
+  componentWillMount() {
+    this.getHomeSettings(this.props)
+  }
+
+  handleHomeAppChange = (index, apps) => {
+    this.setState({ selectedHomeAppName: apps[index] })
+  }
+
+  handleHomeNameChange = event => {
+    this.setState({
+      homeAppName: event.target.value && event.target.value.trim(),
+    })
+  }
+
+  handleHomeSettingsSave = async () => {
+    const { walletWeb3, apps, account } = this.props
+    const { storageApp, selectedHomeAppName, homeAppName } = this.state
+
+    if (storageApp && storageApp.proxyAddress) {
+      const selectedAppAddr = apps.find(
+        ({ name }) => name === selectedHomeAppName
+      )
+
+      try {
+        if (selectedHomeAppName !== defaultName) {
+          await AragonStorage.set(
+            walletWeb3,
+            storageApp.proxyAddress,
+            account,
+            'HOME_APP_NAME',
+            homeAppName
+          )
+        }
+
+        await AragonStorage.set(
+          walletWeb3,
+          storageApp.proxyAddress,
+          account,
+          'HOME_APP',
+          (selectedAppAddr && selectedAppAddr.proxyAddress) || ''
+        )
+        window.location.reload(true)
```

Comment:
It is better to avoid this reload, try to listen for event changes instead, since an event is emitted after calling the contract function

Reply:
Yes this is because we are not handling state from the event of the contract, and passing it to state. I am working on it.